### PR TITLE
Create new conversation

### PR DESCRIPTION
**Pull request by LUMacKenn**
- Refactored "BackButton" component to be a generic action button (now also used for new conversation)
- Ability to create new conversation from patient side

Known issues:
Sometimes create fails when patient has just been created? Not sure exactly how to reproduce. Will fix in this PR or followup if needed

[sc-7774]

### CODE REVIEW

File: src/features/conversations/hooks/useCreateConversation.hook.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        user_ids: userIds,
+        name,
+      },
+    },
+  } as ChatConversationBody);
+
+  if (response?.body) {
+    return response.body as CreateChatConversationResponse;
+  }
+  throw new Error("An unknown error occurred");
```
Comment (lvancraen) [inline src/features/conversations/hooks/useCreateConversation.hook.tsx:42]: Let's provide a little more context on the error. If you don't want to give too accurate of a message, let's at least say that the error occurred when creating a group chat.

File: src/features/conversations/hooks/useCreateConversation.hook.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+};
+
+export const useCreateConversation = () => {
+  const queryClient = useQueryClient();
+
+  return useMutation(createChatConversation, {
+    onSuccess: () => {
```
Comment (lvancraen) [inline src/features/conversations/hooks/useCreateConversation.hook.tsx:49]: No onError? Or is it simply bubbled up further?

File: src/features/conversations/hooks/useCreateConversationAndSendMessage.hook.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        name,
+      });
+      const conversationSid = createdConversation.conversation_sid;
+
+      const conversation = await twilioClient?.getConversationBySid(
+        conversationSid
+      );
+
+      const response = await sendMessage(conversation)(firstMessage).catch(
+        (e) => console.error("error here", formatError(e))
```
Comment (lvancraen) [inline src/features/conversations/hooks/useCreateConversationAndSendMessage.hook.tsx:34]: I noticed you have already wrapped this part in a try/catch. Any reason why you want to swallow the error here?

File: src/features/conversations/hooks/useCreateConversationAndSendMessage.hook.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      );
+      console.log(response);
```
Comment (lvancraen) [inline src/features/conversations/hooks/useCreateConversationAndSendMessage.hook.tsx:36]: You probably don't want this anymore, right?
```suggestion
```

**Review by lvancraen — APPROVED** [review]
I love the initiative of factorizing code like you did with the ActionButton. I have a few small comments which I think will improve the durability of the code. But overall, lgtm

File: src/features/conversations/hooks/useCreateConversationAndSendMessage.hook.tsx (hunk 3 above)
Comment (LUMacKenn) [reply src/features/conversations/hooks/useCreateConversationAndSendMessage.hook.tsx]: Ah thanks this was from debugging

File: src/features/conversations/hooks/useCreateConversation.hook.tsx (hunk 2 above)
Comment (LUMacKenn) [reply src/features/conversations/hooks/useCreateConversation.hook.tsx:49]: Bubbled up, yup! Want to handle this from the UI. Later when we have some sort of global error handling we can catch it here and log it or something

**Review by lvancraen — APPROVED** [review]

**Pull request merged**